@here/harp-vectortile-datasource/lib/VectorTileDecoder.ts

```diff
@@ -311,21 +264,22 @@ export class VectorTileDecoder extends ThemedTileDecoder {
         }
         const dataAdapter = this.m_dataAdapter!;
         assert(dataAdapter.canProcess(data));
+        if (this.m_optionsChanged) {
+            if (dataAdapter instanceof OmvDataAdapter) {
+                const omvOptions = this.m_options as OmvDecoderOptions;
+                dataAdapter.configure(omvOptions, styleSetEvaluator);
```

Comment:
why not put this method into the DataAdapter interface (Open-closed principle if I'm not mistaken)...?

Reply:
Yes, it's kind of hacky. If we only had a configure method getting an option map, I'd move it to DataAdapter interface. But there's two more things specific of OMV:
1. You have to pass the StyleSetEvaluator to create a StyleSetDataFilter.
2. You have to retrieve back a OmvFeatureFilter created by OmvDataAdapter to use it to filter kinds in VectorTileDataProcessor (see a couple of lines below the call to configure).

These details are so far specific of OMV. We could generalize them for all formats, but the question is if we really want to do it:
- Kinds filtering is deprecated (see BaseTechniqueParams.kind), eventually we'll remove it.
- All kinds of filtering could be implemented directly in the style, using dynamic properties if necessary.
- FeatureModifiers where introduced for political views, but I think it should be doable via styling  as well.

I think this needs a revisit for further refactoring, maybe when we add OCM.

I can put a comment and create a ticket for this.

Reply:
@atomicsulfate , you could theoretically pass the `This` pointer, so the data adapter could extract the options / stylesetevaluator internally and not here.

However I get your point, if this is mostly things that can be deprecated, then a comment + ticket should be enough to let future developers no that this shouldn't be permanent.

Reply:
Ticket created